src/runtime_src/core/tools/xbutil2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,134 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "SubCmdConfigure.h"
+#include "OO_HostMem.h"
+
+#include "common/system.h"
+#include "common/device.h"
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream> 
+
+#include "common/system.h"
+#include "common/device.h"
+#include <boost/format.hpp>
+#include <map>
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+SubCmdConfigure::SubCmdConfigure(bool _isHidden, bool _isDepricated, bool _isPreliminary)
+    : SubCmd("configure", 
+             "Device and host configuration")
+{
+  const std::string longDescription = "Device and host configuration.";
+  setLongDescription(longDescription);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+
+void
+SubCmdConfigure::execute(const SubCmdOptions& _options) const
+{
+  // -- Common top level options ---
+  bool help = false;
+
+  po::options_description commonOptions("Common Options"); 
+  commonOptions.add_options()
+    ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+
+  po::options_description hiddenOptions("Hidden Options"); 
+
+  // -- Define the supporting option options ----
+  SubOptionOptions subOptionOptions;
+  subOptionOptions.emplace_back(std::make_shared<OO_HostMem>("host-mem"));
+
+  for (auto & subOO : subOptionOptions) {
+    if (subOO->isHidden()) 
+      hiddenOptions.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    else
+      commonOptions.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    subOO->setExecutable(getExecutableName());
+    subOO->setCommand(getName());
+  }
+
+  po::options_description allOptions("All Options");
+  allOptions.add(commonOptions);
+  allOptions.add(hiddenOptions);
+
+  // =========== Process the options ========================================
+
+  // 1) Process the common top level options 
+  po::parsed_options parsedCommonTop = 
+    po::command_line_parser(_options).
+    options(allOptions).          
+    allow_unregistered().           // Allow for unregistered options
+    run();                          // Parse the options
+
+  po::variables_map vm;
+
+  try {
+    po::store(parsedCommonTop, vm);  // Can throw
+    po::notify(vm);                  // Can throw (but really isn't used)
+
+    // Multual DRC
```

Comment:
Should be refactored into a helper function for it is used in multiple locations.

Reply:
Just the following 3 lines?

Reply:
Yes, the following three lines.  While the algorithm is short, not all folks will know how to create it so creating a helper function would be best.

Reply:
I tried that, but looks like I'll have to change a lot of structure, adding the helper in SubCmdConfigure doesn't work. I would have to add it to SubCmd.h which didn't seem right to me.